src/main/scala/com/wavesplatform/state/diffs/ContractInvocationTransactionDiff.scala

```diff
@@ -80,6 +102,16 @@ object ContractInvocationTransactionDiff {
                     )
                     _ <- Either.cond(true, (), ValidationError.NegativeAmount(-42, "")) //  - sum doesn't overflow
                     _ <- Either.cond(true, (), ValidationError.NegativeAmount(-42, "")) //  - whatever else tranfser/massTransfer ensures
+                    _ <- {
+                      val minWaves = (tx.checkedAssets().count(blockchain.hasAssetScript) +
+                        ps.count(_._3.fold(false)(blockchain.hasAssetScript))) * ScriptExtraFee + FeeConstants(ContractInvocationTransaction.typeId) * FeeUnit
+                      Either.cond(
+                        minWaves <= wavesFee,
+                        (),
+                        GenericError(s"Fee in ${tx.assetFee._1
+                          .fold("WAVES")(_.toString)} for ${tx.builder.classTag} does not exceed minimal value of $minWaves WAVES: ${tx.assetFee._2}")
```

Comment:
We can specifty error better here, explaining why exactly fees are not enough and how many do we expect